functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,230 @@ export const processNewAllotment = functions.database
+                submissionTimestamp: helpers.validateCellValue(row, columnsIndex, 'Timestamp'),
+                submissionSerial: helpers.validateCellValue(row, columnsIndex, 'SubmissionSerial'),
+            };
+
+
+            if (lastFile.file_name === AudioFileName) {
+                if (!skip)
+                    lastFile.chunks.push({
+                        audioFileName: AudioFileName,
+                        beginning: moment.duration(row[columnsIndex['Beginning']]).asSeconds(),
+                        ending: moment.duration(row[columnsIndex['Ending']]).asSeconds(),
+                        continuationFrom,
+                        contentReporting,
+                        importTimestamp: admin.database.ServerValue.TIMESTAMP,
+                        processingResolution: row[columnsIndex['Resolution']],
+                    });
+            } else { // NEW FILE -- Save the previous Chunks and CLEAR
+                if (lastFile.file_name != null && !lastFile.skip) {
+                    summary.addedChunks++;
+
+                    //////////////////////////////////////////
+                    //
+                    // Ensuring chunks do NOT overlap in time
+                    //
+                    //////////////////////////////////////////
+                    let lastEndingTime = null;
+                    for (let k = 0; k < lastFile.chunks.length; k++) {
+                        let chunk = lastFile.chunks[k];
+                        
+                        let beginning = moment(chunk.beginning, "HH:mm:ss");
+                        let trackNameRegex = /\w+-\d+(.*)/;            
+                        let track = trackNameRegex.exec(AudioFileName)[1];
+
+                        // check if last chunck ending time is GREATER than current chunk ending time 
+                        if (lastEndingTime && lastFile.track == track && lastEndingTime.diff(beginning)._data.seconds >= 0 ) {
+                                summary.overlappingChunks.push(lastFile);
+                                console.log('Overlapping chunks');
+                        }
+                        lastEndingTime = moment(chunk.ending, "HH:mm:ss");
+                    }
+
+
+                    ///////////////////////////////
+                    // Sorting by Beginning time
+                    ///////////////////////////////
+                    let chunks = lastFile.chunks;
+
+                    // `start` attribute is added to be used in sorting
+                    chunks.forEach(element => element.start = moment.duration(element.beginning).asSeconds());
+                    chunks = lodash.orderBy(chunks, ['start'], ['asc']);
+
+                    // getting rid of the added `start` attribute
+                    chunks.forEach(element => delete element.start);
+
+
+
+                    ////////////////////////////////////////////////////////
+                    // Filling `continuationFrom` only for the FIRST column
```

Comment:
Why do you use “column” here? Cannot understand it.

Reply:
Moreover, why are you doing this? Is there a requirement in the task for this?

Reply:
I guess why. You misunderstood the “Continuation From column should be filled only for the first chunk of the track (file)” validity check requirement. You should **verify** that the “Continuation From” column is filled only for the first chunk, whereas you just **remove** this value for all the chunks except the first one. This is wrong.

Reply:
And please add `continuationTo` property as described in the data structure. It is not existent in the spreadsheet, you'll have to add it based on the `continuationFrom`.